src/test/java/com/microsoft/azure/vmagent/test/IntegrationTest.java

```diff
@@ -367,19 +374,21 @@ protected boolean areAllVMsDeployed(final List<String> vmNames) throws AzureClou
                 final String resource = op.targetResource().resourceName();
                 final String state = op.provisioningState();
                 if (op.targetResource().resourceType().contains("virtualMachine")) {
-                    if (!state.equalsIgnoreCase("creating")
-                            && !state.equalsIgnoreCase("succeeded")
-                            && !state.equalsIgnoreCase("running")) {
-                        return false;
-                    } else if (state.equalsIgnoreCase("succeeded")) {
-                        for (String vmName : vmNames) {
-                            if (resource.equalsIgnoreCase(vmName)) {
-                                deployedVMs++;
-                                break;
+                    switch (state.toLowerCase()) {
+                        case "creating":
+                        case "running":
```

Comment:
logic changed here. former is when  !creating && !succeeded && !running then return false, now when creating || running return false.

Reply:
They're identical except that the former is obscure and doesn't handle failed case

Reply:
got it.